erpnext/selling/report/sales_analytics/sales_analytics.py

```diff
@@ -276,7 +276,11 @@ def get_supplier_parent_child_map(self):
 
 	def get_chart_data(self):
 		length = len(self.columns)
-		labels = [d.get("label") for d in self.columns[2:length-1]]
+
+		if self.filters.tree_type in ["Customer", "Supplier", "Item"]:
+			labels = [d.get("label") for d in self.columns[2:length-1]]
```

Comment:
You don't need to provide `length-1`. You can list slice by using `self.columns[2:]` as well.

Although, if you're using list slicing, good practice would be to add a comment to say exactly why you are doing this, since it's not exactly clear why you're skipping some columns.

Reply:
using ```self.columns[2:]``` will only slice first two columns where as I need to slice last column as well, thats why I have used ```length-1```.These columns are non-numeric columns which doesn't need to be displayed in graph.

Reply:
`length-1` will not slice the last column. You will need to do `length-2` for that. Or `self.columns[2:-1]`.

And add that description in code as a comment.